src/components/containers/menu/Menu.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ToggleButton from '../../controls/buttons/ToggleButton';
+import MenuPanel from './MenuPanel';
+import MenuSection from './MenuSection';
+
+class Menu extends React.Component {
+  static childContextTypes = {
+    inline: PropTypes.bool
+  };
+
+  state = {
+    isMenuOpen: false
+  };
+
+  getChildContext() {
+    return { inline: this.props.inline };
+  }
+
+  toggleMenu = () => {
+    this.setState({
```

Comment:
When you use `setState` and are relying on previous state, you should use the function argument method instead:

```javascript
this.setState(previousState => ({ isMenuOpen: !previousState.isMenuOpen }))
```

The reason for this is that state changes are queued and not guaranteed to be in order. Although it's unlikely to be a problem in this scenario, it's safer to use the function syntax.

Reply:
Thanks, didn't know that.